From 1279e0c5ca0aa66ee741a22132a6bbe0245cd417 Mon Sep 17 00:00:00 2001
From: Matthias Kuhn <matthias@opengis.ch>
Date: Fri, 4 Jul 2025 09:01:56 +0200
Subject: [PATCH] Do not hardcode vector layer properties dialog icon

---
 src/ui/qgsvectorlayerpropertiesbase.ui | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/ui/qgsvectorlayerpropertiesbase.ui b/src/ui/qgsvectorlayerpropertiesbase.ui
index 4145f3a01ed9..8526f1dc1df5 100644
--- a/src/ui/qgsvectorlayerpropertiesbase.ui
+++ b/src/ui/qgsvectorlayerpropertiesbase.ui
@@ -19,10 +19,6 @@
   <property name="windowTitle">
    <string>Layer Properties</string>
   </property>
-  <property name="windowIcon">
-   <iconset resource="../../images/images.qrc">
-    <normaloff>:/images/icons/qgis-icon-16x16.png</normaloff>:/images/icons/qgis-icon-16x16.png</iconset>
-  </property>
   <layout class="QVBoxLayout" name="verticalLayout_26">
    <item>
     <widget class="QSplitter" name="mOptionsSplitter">
