diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 4a642696c43..f1b75c5de88 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -444,11 +444,9 @@ foreach(module ${PY_MODULES})
   install(FILES ${BINDING_FILES_ROOT_DIR}/${module}/class_map.yaml DESTINATION "${QGIS_PYTHON_DIR}/${module_name}")
 
   # install source sip files
-  if(SIP_GLOBAL_INSTALL)
-    file(GLOB sip_files ${CMAKE_CURRENT_BINARY_DIR}/${module}/*.sip)
-    install(FILES ${sip_files} DESTINATION ${SIP_DEFAULT_SIP_DIR}/qgis/${module})
-    install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${module}/auto_generated DESTINATION ${SIP_DEFAULT_SIP_DIR}/qgis/${module})
-  endif()
+  file(GLOB sip_files ${CMAKE_CURRENT_BINARY_DIR}/${module}/*.sip)
+  install(FILES ${sip_files} DESTINATION ${SIP_DEFAULT_SIP_DIR}/qgis/${module})
+  install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${module}/auto_generated DESTINATION ${SIP_DEFAULT_SIP_DIR}/qgis/${module})
 
   if(SIP_BUILD_EXECUTABLE)
     add_custom_command(TARGET py${module}
