diff --git a/cmake/SIPMacros.cmake b/cmake/SIPMacros.cmake
index a0bfb1e6e22..26a33343780 100644
--- a/cmake/SIPMacros.cmake
+++ b/cmake/SIPMacros.cmake
@@ -199,5 +199,5 @@ MACRO(BUILD_SIP_PYTHON_MODULE MODULE_NAME SIP_FILES EXTRA_OBJECTS)
       )
   ENDIF(WIN32)
 
-  INSTALL(TARGETS ${_logical_name} DESTINATION "${Python_SITEARCH}/${_parent_module_path}")
+  INSTALL(TARGETS ${_logical_name} DESTINATION "${QGIS_PYTHON_DIR}/${_parent_module_path}")
 ENDMACRO(BUILD_SIP_PYTHON_MODULE MODULE_NAME SIP_FILES EXTRA_OBJECTS)
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 0a4b489b908..60561ea8757 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -140,7 +140,7 @@ ELSE()
   SET(QSCI_SIP_MOD_NAME Qsci/qscimod5.sip)
 ENDIF()
 
-set(QGIS_PYTHON_DIR ${Python_SITEARCH}/qgis)
+set(QGIS_PYTHON_DIR "${Python_SITEARCH}/qgis" CACHE PATH "Installation directory for Python bindings")
 
 # core module
 file(GLOB_RECURSE sip_files_core core/*.sip core/*.sip.in)
@@ -424,9 +424,7 @@ foreach(module ${PY_MODULES})
   PY_COMPILE(py${module} "${QGIS_PYTHON_OUTPUT_DIRECTORY}/${module}")
 
   # install source sip files
-  if(SIP_GLOBAL_INSTALL)
-    file(GLOB sip_files ${CMAKE_CURRENT_BINARY_DIR}/${module}/*.sip)
-    install(FILES ${sip_files} DESTINATION ${SIP_DEFAULT_SIP_DIR}/qgis/${module})
-    install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${module}/auto_generated DESTINATION ${SIP_DEFAULT_SIP_DIR}/qgis/${module})
-  endif()
+  file(GLOB sip_files ${CMAKE_CURRENT_BINARY_DIR}/${module}/*.sip)
+  install(FILES ${sip_files} DESTINATION ${QGIS_PYTHON_DIR}/bindings/${module})
+  install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${module}/auto_generated DESTINATION ${QGIS_PYTHON_DIR}/bindings/${module})
 endforeach(module)
diff --git a/python/PyQt/CMakeLists.txt b/python/PyQt/CMakeLists.txt
index 09807ef474e..0a2488e47ab 100644
--- a/python/PyQt/CMakeLists.txt
+++ b/python/PyQt/CMakeLists.txt
@@ -1,4 +1,4 @@
-set (QGIS_PYQT_DIR ${Python_SITEARCH}/qgis/PyQt)
+set (QGIS_PYQT_DIR ${QGIS_PYTHON_DIR}/PyQt)
 
 set(PYQT_COMPAT_FILES
   __init__.py
diff --git a/python/processing/CMakeLists.txt b/python/processing/CMakeLists.txt
index 167a7be08a1..9cca052d840 100644
--- a/python/processing/CMakeLists.txt
+++ b/python/processing/CMakeLists.txt
@@ -1,7 +1,5 @@
 # See ../CMakeLists.txt for info on staged-plugins* and clean-staged-plugins targets
 
-set(QGIS_PYTHON_DIR ${Python_SITEARCH}/qgis)
-
 set(PY_FILES
   __init__.py
   algfactory.py
