diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index f785bded73c..83e44bf430c 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -2174,6 +2174,7 @@ set(QGIS_CORE_HDRS
   vectortile/qgsvtpktiles.h
   vectortile/qgsvtpkvectortiledataprovider.h
   vectortile/qgsxyzvectortiledataprovider.h
+  symbology/qgsblanksegmentutils_p.h
 )
 
 set(QGIS_CORE_PRIVATE_HDRS
@@ -2203,7 +2204,6 @@ set(QGIS_CORE_PRIVATE_HDRS
 
   providers/ogr/qgscplerrorhandler_p.h
 
-  symbology/qgsblanksegmentutils_p.h
 
   textrenderer/qgstextrenderer_p.h
 )
