diff --git a/src/app/3d/qgsvectorlayer3drendererwidget.h b/src/app/3d/qgsvectorlayer3drendererwidget.h
index cae345cde66..8daca9e7bf9 100644
--- a/src/app/3d/qgsvectorlayer3drendererwidget.h
+++ b/src/app/3d/qgsvectorlayer3drendererwidget.h
@@ -57,7 +57,7 @@ class QgsSingleSymbol3DRendererWidget : public QWidget
 
 
 //! Widget for configuration of 3D renderer of a vector layer
-class QgsVectorLayer3DRendererWidget : public QgsMapLayerConfigWidget
+class APP_EXPORT QgsVectorLayer3DRendererWidget : public QgsMapLayerConfigWidget
 {
     Q_OBJECT
   public:
@@ -86,7 +86,7 @@ class QgsVectorLayer3DRendererWidget : public QgsMapLayerConfigWidget
     QgsRuleBased3DRendererWidget *widgetRuleBasedRenderer = nullptr;
 };
 
-class QgsVectorLayer3DRendererWidgetFactory : public QObject, public QgsMapLayerConfigWidgetFactory
+class APP_EXPORT QgsVectorLayer3DRendererWidgetFactory : public QObject, public QgsMapLayerConfigWidgetFactory
 {
     Q_OBJECT
   public:
