diff --git a/src/native/CMakeLists.txt b/src/native/CMakeLists.txt
index 8631b5ae49d..467abd10aff 100644
--- a/src/native/CMakeLists.txt
+++ b/src/native/CMakeLists.txt
@@ -106,6 +106,10 @@ if (UNIX AND NOT APPLE AND NOT ANDROID)
   target_link_libraries(qgis_native ${QT_VERSION_BASE}::DBus)
 endif()
 
+if(APPLE)
+  target_link_libraries(qgis_native "-framework AppKit")
+endif()
+
 # install
 if(NOT QGIS_MAC_BUNDLE)
   install(FILES ${QGIS_NATIVE_HDRS} DESTINATION ${QGIS_INCLUDE_DIR})
