diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index cd5034fc15d..91449f8a4e6 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -961,6 +961,7 @@ set(QGIS_GUI_HDRS
   qgsmaplayerconfigwidget.h
   qgsmaplayerconfigwidgetfactory.h
   qgsmaplayersavestyledialog.h
+  qgsmaplayerserverpropertieswidget.h
   qgsmaplayerloadstyledialog.h
   qgsmaplayerrefreshsettingswidget.h
   qgsmaplayerstylecategoriesmodel.h
