diff --git a/src/3d/terrain/qgsterraingenerator.cpp b/src/3d/terrain/qgsterraingenerator.cpp
index 6e7e85c46c3..4189117a0cb 100644
--- a/src/3d/terrain/qgsterraingenerator.cpp
+++ b/src/3d/terrain/qgsterraingenerator.cpp
@@ -42,7 +42,7 @@ void QgsTerrainGenerator::rootChunkHeightRange( float &hMin, float &hMax ) const
 {
   // TODO: makes sense to have kind of default implementation?
   hMin = 0;
-  hMax = 400;
+  hMax = 8848;
 }
 
 float QgsTerrainGenerator::heightAt( double x, double y, const Qgs3DRenderContext &context ) const
