diff --git a/i18n/qgis_de.ts b/i18n/qgis_de.ts
index 96c65184331..9687cb05a92 100644
--- a/i18n/qgis_de.ts
+++ b/i18n/qgis_de.ts
@@ -140531,13 +140531,13 @@ Fehlermeldung: %1</translation>
     <name>QgsO2</name>
     <message>
         <location filename="../src/auth/oauth2/core/qgso2.cpp" line="182"/>
-        <source>QGIS OAuth2 verification has finished.</source>
-        <translation>QGIS-OAuth2-Verifikation abgeschlossen.</translation>
+        <source>%1 OAuth2 verification has finished.</source>
+        <translation>%1-OAuth2-Verifikation abgeschlossen.</translation>
     </message>
     <message>
         <location filename="../src/auth/oauth2/core/qgso2.cpp" line="183"/>
-        <source>You can close this window and return to QGIS.</source>
-        <translation>Das Fenster kann geschlossen und zu QGIS zurückgekehrt werden.</translation>
+        <source>You can close this window and return to %1.</source>
+        <translation>Das Fenster kann geschlossen und zu %1 zurückgekehrt werden.</translation>
     </message>
 </context>
 <context>
@@ -192745,4 +192745,4 @@ The algorithm also makes it possible to reorder the bands for the newly-created
         <translation>Drittes Quartil (Q3)</translation>
     </message>
 </context>
-</TS>
\ No newline at end of file
+</TS>
diff --git a/i18n/qgis_en_US.ts b/i18n/qgis_en_US.ts
index 87dd634b253..9123297ff01 100644
--- a/i18n/qgis_en_US.ts
+++ b/i18n/qgis_en_US.ts
@@ -140605,13 +140605,13 @@ Error message: %1</translation>
     <name>QgsO2</name>
     <message>
         <location filename="../src/auth/oauth2/core/qgso2.cpp" line="182"/>
-        <source>QGIS OAuth2 verification has finished.</source>
-        <translation>QGIS OAuth2 verification has finished.</translation>
+        <source>%1 OAuth2 verification has finished.</source>
+        <translation>%1 OAuth2 verification has finished.</translation>
     </message>
     <message>
         <location filename="../src/auth/oauth2/core/qgso2.cpp" line="183"/>
-        <source>You can close this window and return to QGIS.</source>
-        <translation>You can close this window and return to QGIS.</translation>
+        <source>You can close this window and return to %1.</source>
+        <translation>You can close this window and return to %1.</translation>
     </message>
 </context>
 <context>
@@ -192822,4 +192822,4 @@ The algorithm also makes it possible to reorder the bands for the newly-created
         <translation>Third Quartile (Q3)</translation>
     </message>
 </context>
-</TS>
\ No newline at end of file
+</TS>
diff --git a/i18n/qgis_fr.ts b/i18n/qgis_fr.ts
index e99a43569c5..109c67874cf 100644
--- a/i18n/qgis_fr.ts
+++ b/i18n/qgis_fr.ts
@@ -140242,12 +140242,12 @@ Message d&apos;erreur : %1</translation>
     <name>QgsO2</name>
     <message>
         <location filename="../src/auth/oauth2/core/qgso2.cpp" line="182"/>
-        <source>QGIS OAuth2 verification has finished.</source>
+        <source>%1 OAuth2 verification has finished.</source>
         <translation type="unfinished"/>
     </message>
     <message>
         <location filename="../src/auth/oauth2/core/qgso2.cpp" line="183"/>
-        <source>You can close this window and return to QGIS.</source>
+        <source>You can close this window and return to %1.</source>
         <translation type="unfinished"/>
     </message>
 </context>
@@ -192383,4 +192383,4 @@ Chaque élément définit une &quot;map&quot; avec les clés suivantes:&lt;/p&gt
         <translation>Troisième Quartile (Q3)</translation>
     </message>
 </context>
-</TS>
\ No newline at end of file
+</TS>
diff --git a/i18n/qgis_it.ts b/i18n/qgis_it.ts
index a581b81a038..09963385bed 100644
--- a/i18n/qgis_it.ts
+++ b/i18n/qgis_it.ts
@@ -140353,13 +140353,13 @@ Messaggio errore: %1</translation>
     <name>QgsO2</name>
     <message>
         <location filename="../src/auth/oauth2/core/qgso2.cpp" line="182"/>
-        <source>QGIS OAuth2 verification has finished.</source>
+        <source>%1 OAuth2 verification has finished.</source>
         <translation type="unfinished"/>
     </message>
     <message>
         <location filename="../src/auth/oauth2/core/qgso2.cpp" line="183"/>
-        <source>You can close this window and return to QGIS.</source>
-        <translation>Puoi chiudere questa finestra e tornare a QGIS.</translation>
+        <source>You can close this window and return to %1.</source>
+        <translation>Puoi chiudere questa finestra e tornare a %1.</translation>
     </message>
 </context>
 <context>
@@ -192535,4 +192535,4 @@ L&apos;algoritmo rende inoltre possibile riordinare le bande del nuovo raster.</
         <translation>Terzo Quartile (Q3)</translation>
     </message>
 </context>
-</TS>
\ No newline at end of file
+</TS>
diff --git a/src/auth/oauth2/core/qgso2.cpp b/src/auth/oauth2/core/qgso2.cpp
index d732d878b8a..82921d46e4f 100644
--- a/src/auth/oauth2/core/qgso2.cpp
+++ b/src/auth/oauth2/core/qgso2.cpp
@@ -23,6 +23,7 @@
 #include "qgsnetworkaccessmanager.h"
 #include "qgssetrequestinitiator_p.h"
 
+#include <QCoreApplication>
 #include <QCryptographicHash>
 #include <QDir>
 #include <QJsonDocument>
@@ -179,8 +180,8 @@ void QgsO2::setVerificationResponseContent()
   {
     setReplyContent(
       QString::fromUtf8( verhtml.readAll() )
-        .replace( "{{ H2_TITLE }}"_L1, tr( "QGIS OAuth2 verification has finished." ) )
-        .replace( "{{ H3_TITLE }}"_L1, tr( "You can close this window and return to QGIS." ) )
+        .replace( "{{ H2_TITLE }}"_L1, tr( "%1 OAuth2 verification has finished." ).arg( QCoreApplication::applicationName() ) )
+        .replace( "{{ H3_TITLE }}"_L1, tr( "You can close this window and return to %1." ).arg( QCoreApplication::applicationName() ) )
         .toUtf8()
     );
   }
