diff --git a/cmake/FindSIP.cmake b/cmake/FindSIP.cmake
index 1234567..abcdefg 100644
--- a/cmake/FindSIP.cmake
+++ b/cmake/FindSIP.cmake
@@ -38,7 +38,9 @@ ELSE(SIP_VERSION)
     STRING(REGEX REPLACE "^sip_version:([^\n]+).*$" "\\1" SIP_VERSION ${sip_config})
     STRING(REGEX REPLACE ".*\nsip_version_num:([^\n]+).*$" "\\1" SIP_VERSION_NUM ${sip_config})
     STRING(REGEX REPLACE ".*\nsip_version_str:([^\n]+).*$" "\\1" SIP_VERSION_STR ${sip_config})
-    STRING(REGEX REPLACE ".*\ndefault_sip_dir:([^\n]+).*$" "\\1" SIP_DEFAULT_SIP_DIR ${sip_config})
+    IF(NOT DEFINED SIP_DEFAULT_SIP_DIR)
+      STRING(REGEX REPLACE ".*\ndefault_sip_dir:([^\n]+).*$" "\\1" SIP_DEFAULT_SIP_DIR ${sip_config})
+    ENDIF()
     IF(NOT SIP_FIND_QUIETLY)
       MESSAGE(STATUS "Found SIP version: ${SIP_VERSION_STR}")
     ENDIF(NOT SIP_FIND_QUIETLY)
