diff --git a/src/core/qgsapplication.cpp b/src/core/qgsapplication.cpp
index 92dcc608b77..ea575e339be 100644
--- a/src/core/qgsapplication.cpp
+++ b/src/core/qgsapplication.cpp
@@ -1203,6 +1203,7 @@ QString QgsApplication::iconsPath()
 
 QString QgsApplication::srsDatabaseFilePath()
 {
+  QString path;
   if ( ABISYM( mRunningFromBuildDir ) )
   {
     QString tempCopy = QDir::tempPath() + "/srs6.db";
@@ -1216,12 +1217,16 @@ QString QgsApplication::srsDatabaseFilePath()
       }
     }
 
-    return tempCopy;
+    path = tempCopy;
   }
   else
   {
-    return pkgDataPath() + QStringLiteral( "/resources/srs.db" );
+    path = pkgDataPath() + QStringLiteral( "/resources/srs.db" );
   }
+  if ( !QFile( path ).exists() )
+    qWarning() << QString( "srs.db file expected at '%1' was not found." ).arg( path );
+
+  return path;
 }
 
 void QgsApplication::setSvgPaths( const QStringList &svgPaths )
